handler_models.go

```diff
@@ -235,37 +235,24 @@ func ModelOwnerVersionZip(owner, name string, user *users.User, tx *gorm.DB,
 	if !valid {
 		modelVersion = ""
 	}
-
-	model, zipPath, ver, em := (&models.Service{}).DownloadZip(r.Context(), tx,
+	svc := &models.Service{Storage: globals.Storage}
+	_, link, _, em := svc.DownloadZip(r.Context(), tx,
 		owner, name, modelVersion, user, r.UserAgent())
 	if em != nil {
 		return nil, em
 	}
 
-	zipFileName := fmt.Sprintf("model-%sv%d.zip", *model.UUID, ver)
-
-	// Remove request header to always serve fresh
-	r.Header.Del("If-Modified-Since")
-	// Set zip response headers
-	w.Header().Set("Content-Type", "application/zip")
-	w.Header().Set("Content-Disposition", fmt.Sprintf("attachment; filename=\"%s\"", zipFileName))
-	_, err := writeIgnResourceVersionHeader(strconv.Itoa(ver), w, r)
-	if err != nil {
-		return nil, gz.NewErrorMessageWithBase(gz.ErrorUnexpected, err)
-	}
-
 	// commit the DB transaction
 	// Note: we commit the TX here on purpose, to be able to detect DB errors
 	// before writing "data" to ResponseWriter. Once you write data (not headers)
-	// into it the status code is set to 200 (OK).
+	// into it the status code is set to 302 (Found).
 	if err := tx.Commit().Error; err != nil {
 		return nil, gz.NewErrorMessageWithBase(gz.ErrorZipNotAvailable, err)
 	}
 
```

Comment:
```suggestion

	r.Header.Del("Authorization")
```

Reply:
Urgh, thanks Github UI. No idea why it made a commit. Sorry about that @marcoshuck , I'll revert the change I added.

+1 to this - Browsers automatically pass through the headers even if the location is a different host.

To this, we should add `r.Header.Del("Private-Token")` (check the private token header to see it matches correctly)